testsuite/test_serverclass.py

```diff
@@ -0,0 +1,267 @@
+
+from pywbem_mock import FakedWBEMConnection
+from dmtf_mof_schema_def import install_dmtf_schema, SCHEMA_MOF_DIR
+
+TEST_SCHEMA = os.path.join(SCHEMA_MOF_DIR, 'test_schema1.mof')
+
+VERBOSE = True
+
+
+class BaseMethodsForTests(object):
+    """
+    Common methods for test of server class.  This includes methods to
+    build the DMTF schema and to build individual instances.
+    """
+
+    def build_schema_list(self, conn, namespace):
+        """
+        Build the schema qualifier and class objects in the repository.
+        This requires only that the leaf objects be defined in a mof
+        include file since the compiler finds the files for qualifiers
+        and dependent classes.
+        TODO: ks 3/18 Right now we must build a file in the schema directory
+        to do this because the compile_from_string does not support
+        include files, etc. It does not search for files. See issue 1138
+        """
+        install_dmtf_schema()
+        class_list = """
+            #pragma locale ("en_US")
+            #pragma include ("Interop/CIM_RegisteredProfile.mof")
+            #pragma include ("Interop/CIM_Namespace.mof")
+            #pragma include ("Interop/CIM_ObjectManager.mof")
+            """
+        # See issue 1138.
+        # conn.compile_mof_str(class_list, namespace=namespace,
+        #                      search_paths=SCHEMA_MOF_DIR)
+        with open(TEST_SCHEMA, "w") as schema_file:
+            schema_file.write(class_list)
+
+        conn.compile_mof_file(TEST_SCHEMA, namespace=namespace,
+                              search_paths=[SCHEMA_MOF_DIR])
+
+        if os.path.isfile(TEST_SCHEMA):
+            os.remove(TEST_SCHEMA)
+        return
+
+    def inst_from_class(self, conn, cls_name, namespace=None, PropertyList=None,
+                        PropertyValues=None, IncludeNullProperties=True,
+                        strict=False, IncludePath=True):
+        """
+        Build an instance from the input classname.
+
+        Parameters:
+          conn (:class:`~pywbem.WBEMConnection`):
+            The WBEMConnection used to get the class from which
+            properties are derived for the instance.
+
+          cls_name
+            String with the class name from which properties are to
+            be derived for the instance.
+
+          namespace (:term:`string`):
+            Namespace in the WBEMConnection used to retrieve the class or
+            `None` if the default_namespace is to be used.
+
+          PropertyList (string or iterable of string)
+            Limits the properties retrieved witht the class and therefore
+            the properties in the resulting instance.
+
+          PropertyValues (dictionary):
+            Dictionary containing nane/value pairs where the names are the
+            names of properties in the class and the properties are the
+            property values to be set into the instance.
```

Comment:
The description leaves it open what happens when PropertyList and PropertyValues do not specify the same set of properties. If we think this is always the same set, PropertyList could be eliminated as a parameter and calculated internally as the set of keys of PropertyValues.

Reply:
1. Modified the code so that we now have a inst_from_class which could at some point be inserted into CIMInstance as from_class and inst_from_classname which just gets the class from the mock repo and calls inst_from_class and is intended just for this test code.

Changed doc to state that if a property is in the PropertyValues but not in the class a ValueError is raised.

I don't think is is always the same set since the PropertyList would be a way to force elimination of properties from the instance even if they had a default value.

2. So now, PropertyList is probably just a way to clip off properties for some tests.  Actually not sure I have a use for it.  I states in the klass parameter definition of inst_from_class that the class should include the properties that the user wants in the instance and must contain the key properties if the include_path is True and strict is True